### ISSUE TITLE
The "cmpxchg" instruction parsing details does not have an "EAX" implicit register.

### ISSUE THREAD

**Issue by dv0** [aquynh/capstone#1454]
example:
 015F611A    F0:0FB11E                  lock cmpxchg dword ptr[esi], ebx

**Comment by aquynh** [comment]
Which version is this? Can you post the output of cstool here?

Also check with the next branch to see if this was fixed

**Comment by dv0** [comment]
I tested it on version 4.0.1.
cstool cmd:
cstool.exe -dus x32 558BECD1C002C3C1AC9800001000128D8C9800001000E1EFE2EDF00FB11EF00FC70E

output:
> 1a  f0 0f b1 1e                                      lock cmpxchg       dword ptr [esi], ebx
        Prefix:0xf0 0x00 0x00 0x00
        Opcode:0x0f 0xb1 0x00 0x00
        rex: 0x0
        addr_size: 4
        modrm: 0x1e
        disp: 0x0
        sib: 0x0
        op_count: 2
                operands[0].type: MEM
                        operands[0].mem.base: REG = esi
                operands[0].size: 4
                operands[0].access: READ
                operands[1].type: REG = ebx
                operands[1].size: 4
                operands[1].access: READ
        Registers read: esi ebx
        EFLAGS: MOD_AF MOD_CF MOD_SF MOD_ZF MOD_PF MOD_OF

1e  f0 0f c7 0e                                      lock cmpxchg8b     qword ptr [esi]
        Prefix:0xf0 0x00 0x00 0x00
        Opcode:0x0f 0xc7 0x00 0x00
        rex: 0x0
        addr_size: 4
        modrm: 0xe
        disp: 0x0
        sib: 0x0
        op_count: 1
                operands[0].type: MEM
                        operands[0].mem.base: REG = esi
                operands[0].size: 8
                operands[0].access: READ
        Registers read: eax ebx ecx edx esi
        Registers modified: eax edx eflags
        EFLAGS: MOD_ZF

**Comment by aquynh** [comment]
fixed now in both branch "v4" (to be released as v4.0.2 soon), and branch "next" (5.0)

```
$ cstool -d x32 F00FB11E 
 0  f0 0f b1 1e                                      lock cmpxchg	dword ptr [esi], ebx
	Prefix:0xf0 0x00 0x00 0x00 
	Opcode:0x0f 0xb1 0x00 0x00 
	rex: 0x0
	addr_size: 4
	modrm: 0x1e
	disp: 0x0
	sib: 0x0
	op_count: 2
		operands[0].type: MEM
			operands[0].mem.base: REG = esi
		operands[0].size: 4
		operands[0].access: READ
		operands[1].type: REG = ebx
		operands[1].size: 4
		operands[1].access: READ
	Registers read: eax esi ebx
	EFLAGS: MOD_AF MOD_CF MOD_SF MOD_ZF MOD_PF MOD_OF
```
